### ISSUE TITLE
Fan-out support

### ISSUE THREAD

**Issue by aschaible** [step-up-labs/firebase-database-dotnet#92]
It would be really nice to have fan-out support as described in this blog entry from the Firebase team:

https://firebase.googleblog.com/2015/10/client-side-fan-out-for-data-consistency_73.html

**Comment by cabauman** [comment]
@bezysoftware let me know if I'm misunderstanding something, but the following works just fine for me:

    public static async Task FanOut<T>(this ChildQuery query, T item, params string[] relativePaths)
    {
        if(relativePaths == null)
        {
            throw new NullReferenceException("realativePaths can't be null.");
        }
        var fanoutObject = new Dictionary<string, T>(relativePaths.Length);
        foreach(var path in relativePaths)
        {
            fanoutObject.Add(path, item);
        }

        await query.PatchAsync(fanoutObject);
    }

    var message = new InboundMessage() { Author = "Colt", Content = "some content" };
    await child.FanOut(message, "myMessages/Message1", "yourMessages/Message1");
    await child.FanOut("Bob", "myMessages/Message1/Author", "yourMessages/Message1/Author");

**Comment by bezysoftware** [comment]
You're right, fan out is just a fancy way of saying multi path update, which can be done with a dictionary. This should be a simple addition to the library
